### PR TITLE
DI-216 Batch Dentist Reporting

### PR DESCRIPTION
**Pull request by SLClayton**
## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-216

## Description

Creates batch reports for comparison of Dentist data from NHS website and chosen DoS DB

### Noteworthy Changes

- No functional changes to main DI project but some functions are changed to be more general use.

## Type of change

Delete not appropriate

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Testing

Please tick the testing that has been completed

- [ ] Unit tests
- [ ] Integration tests

## Developer Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the [code formatting checks](../README.md#code-quality)
- [ ] I have run the [code quality checks](../README.md#code-quality)
- [ ] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [ ] Unit test code coverage is at or above 80%
- [ ] New and existing unit tests pass locally with my changes
- [ ] Tests have added that prove my fix is effective or that my feature works (Integration tests)
- [ ] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

File: Makefile [hunk 1]
```diff
@@ -167,6 +167,23 @@ integration-test: #End to end test DI project - mandatory: PROFILE, TAGS=[comple
...
 		"
 
+create-dentist-reports:
+	make -s docker-run-tools \
+	IMAGE=$$(make _docker-get-reg)/tester:latest \
+	CMD="python comparison_reporting/run_dentist_reports.py" \
+	DIR=./application \
+	ARGS=" \
+		-e PYTHONPATH="." \
+		-e DB_SERVER=$(DB_SERVER) \
```
Comment (rohith1122) [inline Makefile:177]: I believe this should be 
-e DB_SERVER=$(DB_SERVER_NAME) \

File: Makefile [hunk 2]
```diff
@@ -167,6 +167,23 @@ integration-test: #End to end test DI project - mandatory: PROFILE, TAGS=[comple
...
+		-e DB_PORT=$(DB_PORT) \
+		-e DB_NAME=$(DB_NAME) \
+		-e DB_USER_NAME=$(DB_USER_NAME) \
```
Comment (rohith1122) [inline Makefile:180]: -e DB_USER_NAME=$$(make -s secret-get-existing-value NAME=$(DB_USER_NAME_SECRET_NAME) KEY=$(DB_USER_NAME_SECRET_KEY)) \

**Review by rohith1122 — COMMENTED** [review]
other than 2 small changes in Makefile all looking good.

File: Makefile (hunk 1 above)
Comment (SLClayton) [reply Makefile]: As discussed in call I have changed this to 
`-e DB_SERVER=$$(make -s aws-rds-describe-instance-value DB_INSTANCE=$(DB_SERVER_NAME) KEY_DOT_PATH=Endpoint.Address)`

**Pull request merged**